R/download_vpfiles.R

```diff
@@ -1,109 +1,99 @@
-    error = function(e) {
-      print(paste("no data available at URL", url))
-    }
-  )
-  return(z)
-}
-
-#' Download vertical profile files (\code{vp}) from the ENRAM data repository
-#'
-#' Download a set of vp bird profiles from the ENRAM repository. These are
-#' stored within monthly available zip folders. This function downloads and
-#' unzips them at a user defined location. Check
-#' \href{http://enram.github.io/data-repository/}{http://enram.github.io/data-repository/}
-#' for an overview of available data.
-#'
-#' @param date_min ISO fomat date indicating the first date to download
-#' files from.
-#' @param date_max ISO fomat date indicating the last date to download
-#' files from
-#' @param country Char vector with two letter country shortcuts.
-#' @param radar Char vector with three letter radar sindicators. Make sure the
-#' radars selected are in accordance to the country selection
-#' @param directory Char defining the location to store the downloaded zip
-#' folders and unzip into the default folder structure
+#' @param date_min character. YYYY-MM-DD start date of file selection. Days will
+#'   be ignored.
+#' @param date_max character. YYYY-MM-DD end date of file selection. Days will
+#'   be ignored.
+#' @param radars character (vector). 5-letter country/radar code(s)
+#'   (e.g. "bejab") of radars to include in file selection.
+#' @param directory character. Path to local directory where files should be
+#'   downloaded and unzipped.
+#' @param overwrite logical. TRUE for redownloading and overwriting previously
+#'   downloaded files of the same names.
 #'
 #' @export
-#' @importFrom lubridate as_date floor_date
-#' @importFrom curl curl_download
+#' @importFrom curl curl_fetch_disk
 #'
 #' @examples
-#' my_path <- "~/my/directory/"
+#' my_path <- "~/my/directory"
+#' # Download data from radars "bejab" and "bewid", even if previously
+#' # downloaded (overwrite = TRUE). Will successfully download 2016-10 files,
+#' # but show 404 error for 2016-11 files (as these are not available).
 #' \dontrun{
-#' download_vpfiles("2016-10-01", "2016-11-30", c("be"),
-#'   c("jab", "wid"),
-#'   directory = my_path
+#' download_vpfiles(
+#'   date_min = "2016-10-01",
+#'   date_max = "2016-11-30",
+#'   radar = c("bejab", "bewid"),
+#'   directory = my_path,
+#'   overwrite = TRUE
 #' )
 #' }
-download_vpfiles <- function(date_min, date_max, country, radar,
-                             directory = ".") {
-  # create date range set of potential downloadable zip files (if all data
-  # would exist)
-  start <- floor_date(as_date(date_min, tz = NULL), "month")
-  end <- floor_date(as_date(date_max, tz = NULL), "month")
-  dates_to_check <- seq(start, end, by = "months")
+download_vpfiles <- function(date_min, date_max, radars, directory = ".",
+                             overwrite = FALSE) {
+  # Stop if radar codes don't contain exactly 5 characters
+  wrong_codes <- radars[nchar(radars) != 5]
+  if (length(wrong_codes) > 0) {
```

Comment:
Do we have an overview list of the current radars and countries (e.g. compiled from the [overview list](https://lw-enram.s3-eu-west-1.amazonaws.com/radars.csv) to check the existence of radars and countries? Currently, every 5-string combination will be attempted for download.

Reply:
> Currently, every 5-string combination will be attempted for download.

Every 5-letter combination **that the user provides** will be attempted for download, so they are in control. I agree it would be nice to know which radars exist, but so does information about their coverage (for which you also get 404s).

I would suggest to bundle this at some point in a separate function `coverage_vpfiles()` that could provide such information for remote or even downloaded vp files. @CeciliaNilsson709 has already created some nice charts to express the coverage. 👌 And the returned information (list of vp files that are available) could optionally be passed to the `download_vpfiles()` for more targeted download. So something for a separate enhancement issue I would say.